a3fe/configuration/engine_config.py

```diff
@@ -5,298 +5,243 @@
+            "RESTRAIN": [0.0, 1.0],
+            "DISCHARGE": [0.0, 0.291, 0.54, 0.776, 1.0],
+            "VANISH": [
+                0.0, 0.026, 0.054, 0.083, 0.111, 0.14, 0.173, 0.208,
+                0.247, 0.286, 0.329, 0.373, 0.417, 0.467, 0.514,
+                0.564, 0.623, 0.696, 0.833, 1.0,
+            ],
+        },
+        "FREE": {
+            "DISCHARGE": [0.0, 0.222, 0.447, 0.713, 1.0],
+            "VANISH": [
+                0.0, 0.026, 0.055, 0.09, 0.126, 0.164, 0.202, 0.239,
+                0.276, 0.314, 0.354, 0.396, 0.437, 0.478, 0.518,
+                0.559, 0.606, 0.668, 0.762, 1.0,
+            ],
+        },
+    }   
+    return default_values[leg][stage]
 
 class SomdConfig(_EngineRunnerConfig, _BaseModel):
     """
     Pydantic model for holding SOMD engine configuration.
-    """
-    
+    """ 
+    default_lambda_values: _Dict[str, _Dict[str, _List[float]]] = _Field(
+        default={
+            "BOUND": {
+                "RESTRAIN": [0.0, 1.0],
+                "DISCHARGE": [0.0, 0.291, 0.54, 0.776, 1.0],
+                "VANISH": [
+                    0.0, 0.026, 0.054, 0.083, 0.111, 0.14, 0.173, 0.208,
+                    0.247, 0.286, 0.329, 0.373, 0.417, 0.467, 0.514,
+                    0.564, 0.623, 0.696, 0.833, 1.0,
+                ],
+            },
+            "FREE": {
+                "DISCHARGE": [0.0, 0.222, 0.447, 0.713, 1.0],
+                "VANISH": [
+                    0.0, 0.026, 0.055, 0.09, 0.126, 0.164, 0.202, 0.239,
+                    0.276, 0.314, 0.354, 0.396, 0.437, 0.478, 0.518,
+                    0.559, 0.606, 0.668, 0.762, 1.0,
+                ],
+            },
+        },
+        description="Default lambda values for each leg and stage type"
+    )
     ### Integrator - ncycles modified as required by a3fe ###
     nmoves: int = _Field(25000, description="Number of moves per cycle")
     timestep: float = _Field(4.0, description="Timestep in femtoseconds(fs)")
-    runtime: _Union[int, float] = _Field(..., description="Runtime in nanoseconds(ns)")
+    runtime: _Union[int, float] = _Field(5.0, description="Runtime in nanoseconds(ns)")
 
-    input_dir: str = _Field(..., description="Input directory containing simulation config files")
-    @staticmethod
-    def _calculate_ncycles(runtime: float, time_per_cycle: float) -> int:
-        """
-        Calculate the number of cycles given a runtime and time per cycle.
+    ### Constraints ###
+    constraint: str = _Field("hbonds", description="Constraint type, must be hbonds or allbonds")
+    hydrogen_mass_factor: float = _Field(3.0, alias="hydrogen mass repartitioning factor", description="Hydrogen mass repartitioning factor")
+    integrator: _Literal["langevinmiddle", "leapfrogverlet"] = _Field("langevinmiddle", description="Integration algorithm")
+    
+    ### Thermostatting already handled by langevin integrator
+    thermostat: bool = _Field(False, description="Enable thermostat")
```

Comment:
Please increase the description to state that it's not required if using langevin integrator. Thanks